components/type_info.rst

```diff
@@ -90,7 +92,74 @@ that need a simple way to describe a class or anything with a type::
 
 Each of these calls will return you a ``Type`` instance that corresponds to the
 static method used. You can also resolve types from a string (as shown in the
-``bool`` parameter of the previous example)
+``bool`` parameter of the previous example).
+
+
+The TypeInfo component provides a new array shape type to define exact array structures with specific key-value type relationships.
+
+.. versionadded:: 7.3
+
+    The array shape type was introduced in Symfony 7.3.
+
+Array shape types support:
+
+* Required and optional keys
+* Nested array shapes
+* Sealed and unsealed shapes (allowing or rejecting extra entries)
+
+Array shapes can be sealed or unsealed:
+
+- ``array{0: int}`` is sealed: it does not accept extra entries.
+- ``array{0: int, ...}`` is unsealed: it accepts extra entries.
+- You can also define the type of extra keys and values: ``array{0: int, ...<string, bool>}``.
+
+This information is stored on the ``ArrayShapeType`` via its ``extraKeyType`` and ``extraValueType`` and validated in its ``accepts()`` method.
+
+Array shapes support associative array definitions::
+
+    use Symfony\Component\TypeInfo\Type;
+
+    // Simple array shape
+    $type = Type::arrayShape([
+        'name' => Type::string(),
+        'age' => Type::int()
+    ]);
+
+    // With optional keys (denoted by "?" suffix)
+    $type = Type::arrayShape([
+        'required_id' => Type::int(),
+        'optional_name' => ['type' => Type::string(), 'optional' => true],
+    ]);
+
+    // Unsealed: allow extra entries not defined above (sealed = false)
+    $type = Type::arrayShape([
+        'id' => Type::int(),
+    ], false);
+
+    // Unsealed with typed extra keys and values (extraKeyType=string, extraValueType=bool)
+    // Equivalent to: array{id: int, ...<string, bool>}
+    $type = Type::arrayShape([
+        'id' => Type::int(),
+    ], false, Type::string(), Type::bool());
+
+
+Additionally, the ``StringTypeResolver`` supports parsing array shape notation::
+
+    use Symfony\Component\TypeInfo\TypeResolver\TypeResolver;
+
+    $resolver = TypeResolver::create();
+
+    // Parse array shape definition
+    $type = $resolver->resolve('array{name: string, age: int}');
+
+    // Equivalent to:
+    Type::arrayShape([
+        'name' => Type::string(),
+        'age' => Type::int()
+    ]);
+
+    $type->accepts(['name' => 'Alice', 'age' => 30]); // true
+    $type->accepts(['name' => 'Alice', 'age' => '30']); // false (wrong age type)
```

Comment:
I'm not sure that we need this section, as the `StringTypeResolver` is supposed to resolve it, as it has been introduced in TypeInfo.